### PR TITLE
Use ServiceAccount for `flux-giantswarm` Kustomizations

### PR DESCRIPTION
**Pull request by kubasobon**

### CODE REVIEW

**Comment by kubasobon** [comment]
It fixes the `patch/status` situation as well!

**flux-system**
```text
{"level":"error","ts":"2022-02-15T12:27:01.760Z","logger":"controller.kustomization","msg":"unable to update status to progressing","reconciler group":"kustomize.toolkit.fluxcd.io","reconciler kind":"Kustomization","name":"collection","namespace":"flux-giantswarm","error":"kustomizations.kustomize.toolkit.fluxcd.io \"collection\" is forbidden: User \"system:serviceaccount:flux-system:kustomize-controller\" cannot patch resource \"kustomizations/status\" in API group \"kustomize.toolkit.fluxcd.io\" in the namespace \"flux-giantswarm\""}
```

**flux-giantswarm**
```
{"level":"info","ts":"2022-02-15T12:27:38.761Z","logger":"controller.kustomization","msg":"Reconciliation finished in 36.796002219s, next run in 10s","reconciler group":"kustomize.toolkit.fluxcd.io","reconciler kind":"Kustomization","name":"collection","namespace":"flux-giantswarm","revision":"master/0c9bb83785a85ad752bed59f995313d2df4ce992"}
```

**Comment by kubasobon** [comment]
Tested on `gauss`.

File: bootstrap/gs-aws-china/gs-aws-china.yaml [hunk 1]
```diff
@@ -11418,5 +11440,5 @@ metadata:
 spec:
   interval: 30s
   ref:
-    branch: migrate-to-flux-test
+    branch: attach-service-account
```
Comment (rossf7) [inline bootstrap/gs-aws-china/gs-aws-china.yaml:11443]: Do we want to switch to this branch?

**Review by rossf7 — APPROVED** [review]
LGTM just a query on the branch

File: bootstrap/gs-aws-china/gs-aws-china.yaml (hunk 1 above)
Comment (kubasobon) [reply bootstrap/gs-aws-china/gs-aws-china.yaml:11443]: It will be reverted before merging, hence the commit msg `revert: switch to test branch`.

**Pull request merged**